### PR TITLE
chore: owner/group filters

### PR DESCRIPTION
**Pull request by kyle-ssg**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

- Allows users to filter features by user and owner.
- Improves styling of table filter items 

https://github.com/Flagsmith/flagsmith/assets/8608314/a736cbce-1560-483f-a5f9-e07258c54760

## How did you test this code?

- Filtered features by owner and user on the features and identities page

### CODE REVIEW

File: frontend/web/components/pages/FeaturesPage.js [hunk 1]
```diff
@@ -381,6 +388,45 @@ const FeaturesPage = class extends Component {
...
+                                            useLocalStorage
+                                            value={this.state.group_owners}
+                                            onChange={(owners) => {
+                                              FeatureListStore.isLoading = true
+                                              this.setState(
+                                                {
+                                                  group_owners: owners,
+                                                },
+                                                this.filter,
+                                              )
```
Comment (zachaysan) [inline frontend/web/components/pages/FeaturesPage.js]:
```suggestion
                                            onChange={(group_owners) => {
                                              FeatureListStore.isLoading = true
                                              this.setState(
                                                {
                                                  group_owners: group_owners,
                                                },
                                                this.filter,
                                              )
```

File: frontend/web/components/pages/UserPage.js [hunk 2]
```diff
@@ -483,6 +491,46 @@ const UserPage = class extends Component {
...
+                                              useLocalStorage
+                                              value={this.state.group_owners}
+                                              onChange={(owners) => {
+                                                FeatureListStore.isLoading = true
+                                                this.setState(
+                                                  {
+                                                    group_owners: owners,
+                                                  },
+                                                  this.filter,
+                                                )
```
Comment (zachaysan) [inline frontend/web/components/pages/UserPage.js]:
```suggestion
                                              onChange={(group_owners) => {
                                                FeatureListStore.isLoading = true
                                                this.setState(
                                                  {
                                                    group_owners: group_owners,
                                                  },
                                                  this.filter,
                                                )
```

File: frontend/web/components/tables/TableOwnerFilter.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  useEffect(() => {
+    const checkLocalStorage = async function () {
+      if (useLocalStorage && !checkedLocalStorage.current) {
+        checkedLocalStorage.current = true
+        const storedValue = await AsyncStorage.getItem(`${projectId}-owners`)
+        if (storedValue) {
+          try {
+            const storedValueObject = JSON.parse(storedValue)
+            onChange(storedValueObject)
+          } catch (e) {}
```
Comment (zachaysan) [inline frontend/web/components/tables/TableOwnerFilter.tsx:57]: Are we intentionally defaulting to a undefined or null return after an exception here?

**Review by zachaysan — APPROVED** [review]
Overall the code looks great, just a couple of minor suggestions

File: frontend/web/components/tables/TableOwnerFilter.tsx (hunk 3 above)
Comment (kyle-ssg) [reply frontend/web/components/tables/TableOwnerFilter.tsx:57]: Yep, neither the try or the catch return anything.

**Pull request merged**